warden/app/legacy.go

```diff
@@ -164,6 +165,10 @@ func (app *App) registerLegacyModules(appOpts servertypes.AppOptions, wasmOpts [
 	}
 
 	availableCapabilities := strings.Join(AllCapabilities(), ",")
+
+	encoders := WardenProtocolCustomEncoder()
+	wasmOpts = append(wasmOpts, wasmkeeper.WithMessageEncoders(&encoders))
```

Comment:
I see from the docs that wasmkeeper exports two methods (that can't be used together): WithMessageEncoder and WithMessageHandler/WithMessageHandlerDecorator.

From a super quick look, it seems that the MessageHandler allows more freedom in handling messages and marshalling responses back to the contract. But I think that for now the MessageEncore will suffice.

Reply:
Indeed, a solution based on MessageHandler offers more freedom but is more complex to implement and maintain.
It seems that a solution based on MessageEncoder is better documented and more widely used.

I assume that the capabilities of MessageEncoder will be quite sufficient. However, if needed in the future, it will be possible to switch to MessageHandler.